In [1]:
import numpy as np
import pandas as pd
import glob
import os
os.getcwd()
os.chdir('/project/time_series')
from load_data_custom import load_data_using_multi_process
from make_roc_curve import make_roc_curve
os.chdir('/project/time_series/cchc0427')
from lightgbm import LGBMClassifier
from sklearn.model_selection import train_test_split
from model import model_lgbm
from sklearn.metrics import accuracy_score, f1_score, roc_auc_score, precision_score, recall_score
import lazy_loader as lazy
samplerate = lazy.load("samplerate")
resampy = lazy.load("resampy")


In [2]:
import glob
fan_path = '/data/time_series/fan/_6'
len_fan_path = len(glob.glob( fan_path + '/*/*/*' ))

# fan_files_6 = glob.glob(fan_path + '/_6/*/*/*')
# fan_files6 = glob.glob(fan_path + '/6/*/*/*')
fan_files_6 = glob.glob( fan_path + '/*/*/*')

In [3]:
target_lst = []
for i in range(1, 16001):
    if ( i < 40 ) or ( i > 13000):
        continue
    if 16000 % i == 0 :
        target_lst.append(i)

target_lst.append(10000)
target_lst.append(13000)

In [4]:
target_lst

[40,
 50,
 64,
 80,
 100,
 125,
 128,
 160,
 200,
 250,
 320,
 400,
 500,
 640,
 800,
 1000,
 1600,
 2000,
 3200,
 4000,
 8000,
 10000,
 13000]

In [1]:
result = []

for target_sr in target_lst:
    print(f'{target_sr}로 진행중입니다.')
    fan_result = load_data_using_multi_process(fan_files_6, sr = 16000, target_sr = target_sr, res_type = 'kaiser_best') # 데이터 target_sr로 부르기
    
    # librosa.resample하게되면 shape이 ( file수, 1, column수 ) squeeze함수는 shape에서 1인부분 제거 1의의미 모르겠음
    fan_squeeze_result = np.squeeze(fan_result) 
    
    fan_result_df = pd.DataFrame(fan_squeeze_result)
    
    data = fan_result_df.iloc[:, :-1]
    target = fan_result_df.iloc[:, -1]
    
    X_train, X_test, y_train, y_test = train_test_split(data, target, test_size=0.2, shuffle=True, stratify=target, random_state=42)
    
    model = model_lgbm(fan_result_df) # using (model.py)
    
    model.fit(X_train, y_train)

    y_pred = model.predict(X_test)
    y_pred_prob = model.predict_proba(X_test)
    
    acc = accuracy_score(y_test, y_pred) 
    recall = recall_score(y_test, y_pred)
    precision = precision_score(y_test, y_pred)
    f1 = f1_score(y_test, y_pred)    
    auc_score = roc_auc_score(y_test, y_pred_prob[ :, 1 ])
    
    result.append([ target_sr, round(acc, 3) ,round(recall, 3), round(precision, 3), round(f1, 3), round(auc_score, 3) ])

result_df = pd.DataFrame(result, columns = [ 'target_sr', 'acc', 'recall', 'precision', 'f1', 'auc_score'])
result_df.to_csv('/project/time_series/cchc0427/id_num/fanfile_6_target_sr.csv', index = False)


NameError: name 'target_lst' is not defined

In [10]:
target_sr_csv = pd.read_csv('fanfile_6_target_sr.csv')


In [11]:
target_sr_csv

target_sr    acc  recall  precision     f1  auc_score
0          40  0.707   0.075      0.297  0.119      0.502
1          50  0.750   0.200      0.584  0.298      0.662
2          64  0.782   0.264      0.757  0.392      0.726
3          80  0.773   0.251      0.705  0.370      0.693
4         100  0.759   0.166      0.690  0.268      0.658
5         125  0.763   0.180      0.716  0.287      0.704
6         128  0.761   0.169      0.714  0.274      0.704
7         160  0.768   0.190      0.757  0.304      0.699
8         200  0.771   0.169      0.847  0.282      0.689
9         250  0.760   0.142      0.764  0.240      0.678
10        320  0.750   0.119      0.660  0.201      0.631
11        400  0.746   0.098      0.644  0.171      0.607
12        500  0.754   0.095      0.824  0.170      0.597
13        640  0.743   0.081      0.632  0.144      0.638
14        800  0.755   0.098      0.829  0.176      0.625
15       1000  0.742   0.071      0.636  0.128      0.585
16       1600  0.741   0.051      0.652  0.094      0.605
17       2000  0.740   0.047      0.636  0.088      0.567
18       3200  0.734   0.020      0.500  0.039      0.556
19       4000  0.730   0.010      0.273  0.020      0.573
20       8000  0.737   0.031      0.600  0.058      0.559
21      10000  0.732   0.017      0.417  0.033      0.539
22      13000  0.736   0.031      0.562  0.058      0.547

noise = 0 일때 : target_sr = 125, res_type = (soxr_lq, kaiser_fast) 에서 f1, auc 모두 제일높음   
noise = 6 일때 : target_sr = (128, 125), res_type = (soxr_lq) 에서 f1, auc 모두 제일높음      
noise = _6 일때 : target_sr = 64, res_type = (soxr_lq) 에서 f1, auc 모두 제일높음      

In [6]:
sampling_lst = ['abs_max(our_custom)',
                'kaiser_best',
                'kaiser_fast',
                'fft',
                'scipy',
                'polyphase',
                'linear',
                'zero_order_hold',
                'sinc_best',
                'sinc_medium' ,
                'sinc_fastest',
                'soxr_vhq',
                'soxr_hq',
                'soxr_mq',
                'soxr_lq',
                'soxr_hq',
                'soxr_qq']
target_sr = 64
result = []

for res_type in sampling_lst:
    print(f'{res_type}로 진행중입니다.')
    fan_result = fan_result = load_data_using_multi_process(fan_files_6, sr = 16000, target_sr = target_sr, res_type = res_type) # 데이터 target_sr로 부르기
    
    fan_result_df = pd.DataFrame(fan_result)
    
    data = fan_result_df.iloc[:, :-1]
    target = fan_result_df.iloc[:, -1]
    
    X_train, X_test, y_train, y_test = train_test_split(data, target, test_size=0.2, shuffle=True, stratify=target, random_state=42)
    
    model = model_lgbm(fan_result_df) # using (model.py)
    
    model.fit(X_train, y_train)

    y_pred = model.predict(X_test)
    y_pred_prob = model.predict_proba(X_test)
    
    acc = accuracy_score(y_test, y_pred) 
    recall = recall_score(y_test, y_pred)
    precision = precision_score(y_test, y_pred)
    f1 = f1_score(y_test, y_pred)    
    auc_score = roc_auc_score(y_test, y_pred_prob[ :, 1 ])
    
    result.append([ res_type, round(acc, 3) ,round(recall, 3), round(precision, 3), round(f1, 3), round(auc_score, 3) ])

result_df = pd.DataFrame(result, columns = [ 'res_type', 'acc', 'recall', 'precision', 'f1', 'auc_score'])
result_df.to_csv('/project/time_series/cchc0427/noise/fan_6_res_type.csv', index = False)


abs_max(our_custom)로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 223.84290599822998 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


kaiser_best로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 11.521564960479736 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


kaiser_fast로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 6.11199951171875 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


fft로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.7524526119232178 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


scipy로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.6175436973571777 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


polyphase로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 3.1201272010803223 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


linear로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.3265647888183594 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


zero_order_hold로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.272361993789673 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


sinc_best로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.38728404045105 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.
sinc_medium로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")
/package/anaconda3.8/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1221: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


--- 걸린시간: 2.3898766040802 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.
sinc_fastest로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")
/package/anaconda3.8/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1221: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


--- 걸린시간: 2.4696271419525146 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.
soxr_vhq로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")
/package/anaconda3.8/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1221: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


--- 걸린시간: 3.3231992721557617 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


soxr_hq로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.8649702072143555 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


soxr_mq로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.9381706714630127 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


soxr_lq로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.5795657634735107 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


soxr_hq로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.6610958576202393 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


soxr_qq로 진행중입니다.

멀티 프로세싱을 시작합니다. 현재 코어 30개 사용 중. 
1.다른 사람과 동시에 돌리지 마세요.
2.함부로 진행중에 중단 하지마세요.
--- 걸린시간: 2.4742491245269775 ---
normal : 4075
abnormal : 1475
scale_pos_weight is 2.76.
Model is ready to running.


/package/anaconda3.8/lib/python3.8/site-packages/lightgbm/engine.py:177: UserWarning: Found `num_iterations` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


In [4]:
res_type_6_csv = pd.read_csv('/project/time_series/cchc0427/noise/fanfile_6_res_type.csv')
res_type6_csv = pd.read_csv('/project/time_series/cchc0427/noise/fanfile6_res_type.csv')
res_type0_csv = pd.read_csv('/project/time_series/cchc0427/noise/fanfile0_res_type.csv')
res_type_csv

res_type    acc  recall  precision     f1  auc_score
0   abs_max(our_custom)  0.741   0.200      0.536  0.291      0.613
1           kaiser_best  0.782   0.264      0.757  0.392      0.726
2           kaiser_fast  0.783   0.275      0.750  0.402      0.726
3                   fft  0.777   0.244      0.742  0.367      0.697
4                 scipy  0.777   0.244      0.742  0.367      0.697
5             polyphase  0.768   0.231      0.694  0.346      0.707
6                linear  0.717   0.058      0.321  0.098      0.530
7       zero_order_hold  0.717   0.058      0.321  0.098      0.530
8             sinc_best  0.734   0.000      0.000  0.000      0.500
9           sinc_medium  0.734   0.000      0.000  0.000      0.500
10         sinc_fastest  0.734   0.000      0.000  0.000      0.500
11             soxr_vhq  0.781   0.241      0.789  0.369      0.708
12              soxr_hq  0.779   0.241      0.772  0.367      0.720
13              soxr_mq  0.784   0.251      0.796  0.381      0.722
14              soxr_lq  0.771   0.268      0.675  0.383      0.718
15              soxr_hq  0.779   0.241      0.772  0.367      0.720
16              soxr_qq  0.720   0.088      0.382  0.143      0.559

In [6]:
res_type0_csv

target_sr    acc  recall  precision     f1  auc_score
0   abs_max(our_custom)  0.811   0.376      0.810  0.514      0.690
1           kaiser_best  0.842   0.444      0.923  0.600      0.867
2           kaiser_fast  0.853   0.485      0.929  0.637      0.873
3                   fft  0.826   0.407      0.870  0.554      0.836
4                 scipy  0.826   0.407      0.870  0.554      0.836
5             polyphase  0.834   0.431      0.888  0.580      0.853
6                linear  0.778   0.231      0.782  0.356      0.689
7       zero_order_hold  0.778   0.231      0.782  0.356      0.689
8             sinc_best  0.838   0.417      0.939  0.577      0.860
9           sinc_medium  0.837   0.431      0.907  0.584      0.867
10         sinc_fastest  0.848   0.461      0.932  0.617      0.868
11             soxr_vhq  0.841   0.441      0.922  0.596      0.862
12              soxr_hq  0.841   0.444      0.916  0.598      0.859
13              soxr_mq  0.842   0.437      0.935  0.596      0.860
14              soxr_lq  0.852   0.478      0.934  0.632      0.875
15              soxr_hq  0.841   0.444      0.916  0.598      0.859
16              soxr_qq  0.769   0.214      0.724  0.330      0.665

In [9]:
target_sr_6_csv = pd.read_csv('/project/time_series/cchc0427/noise/fanfile_6_target_sr.csv')
target_sr6_csv = pd.read_csv('/project/time_series/cchc0427/noise/fanfile6_target_sr.csv')
target_sr0_csv = pd.read_csv('/project/time_series/cchc0427/noise/fanfile0_target_sr.csv')
target_sr_6_csv

target_sr    acc  recall  precision     f1  auc_score
0          40  0.707   0.075      0.297  0.119      0.502
1          50  0.750   0.200      0.584  0.298      0.662
2          64  0.782   0.264      0.757  0.392      0.726
3          80  0.773   0.251      0.705  0.370      0.693
4         100  0.759   0.166      0.690  0.268      0.658
5         125  0.763   0.180      0.716  0.287      0.704
6         128  0.761   0.169      0.714  0.274      0.704
7         160  0.768   0.190      0.757  0.304      0.699
8         200  0.771   0.169      0.847  0.282      0.689
9         250  0.760   0.142      0.764  0.240      0.678
10        320  0.750   0.119      0.660  0.201      0.631
11        400  0.746   0.098      0.644  0.171      0.607
12        500  0.754   0.095      0.824  0.170      0.597
13        640  0.743   0.081      0.632  0.144      0.638
14        800  0.755   0.098      0.829  0.176      0.625
15       1000  0.742   0.071      0.636  0.128      0.585
16       1600  0.741   0.051      0.652  0.094      0.605
17       2000  0.740   0.047      0.636  0.088      0.567
18       3200  0.734   0.020      0.500  0.039      0.556
19       4000  0.730   0.010      0.273  0.020      0.573
20       8000  0.737   0.031      0.600  0.058      0.559
21      10000  0.732   0.017      0.417  0.033      0.539
22      13000  0.736   0.031      0.562  0.058      0.547

In [15]:
sr_result = pd.concat([target_sr_6_csv, target_sr0_csv, target_sr6_csv], axis = 1)
sr_result

target_sr    acc  recall  precision     f1  auc_score  target_sr    acc  \
0          40  0.707   0.075      0.297  0.119      0.502         40  0.714   
1          50  0.750   0.200      0.584  0.298      0.662         50  0.760   
2          64  0.782   0.264      0.757  0.392      0.726         64  0.794   
3          80  0.773   0.251      0.705  0.370      0.693         80  0.799   
4         100  0.759   0.166      0.690  0.268      0.658        100  0.785   
5         125  0.763   0.180      0.716  0.287      0.704        125  0.842   
6         128  0.761   0.169      0.714  0.274      0.704        128  0.827   
7         160  0.768   0.190      0.757  0.304      0.699        160  0.824   
8         200  0.771   0.169      0.847  0.282      0.689        200  0.808   
9         250  0.760   0.142      0.764  0.240      0.678        250  0.817   
10        320  0.750   0.119      0.660  0.201      0.631        320  0.795   
11        400  0.746   0.098      0.644  0.171      0.607        400  0.779   
12        500  0.754   0.095      0.824  0.170      0.597        500  0.773   
13        640  0.743   0.081      0.632  0.144      0.638        640  0.789   
14        800  0.755   0.098      0.829  0.176      0.625        800  0.789   
15       1000  0.742   0.071      0.636  0.128      0.585       1000  0.789   
16       1600  0.741   0.051      0.652  0.094      0.605       1600  0.785   
17       2000  0.740   0.047      0.636  0.088      0.567       2000  0.781   
18       3200  0.734   0.020      0.500  0.039      0.556       3200  0.777   
19       4000  0.730   0.010      0.273  0.020      0.573       4000  0.775   
20       8000  0.737   0.031      0.600  0.058      0.559       8000  0.777   
21      10000  0.732   0.017      0.417  0.033      0.539      10000  0.777   
22      13000  0.736   0.031      0.562  0.058      0.547      13000  0.779   

    recall  precision     f1  auc_score  target_sr    acc  recall  precision  \
0    0.119      0.376  0.180      0.500         40  0.714   0.122      0.379   
1    0.261      0.616  0.367      0.695         50  0.786   0.322      0.720   
2    0.319      0.770  0.451      0.739         64  0.793   0.325      0.756   
3    0.322      0.805  0.460      0.756         80  0.786   0.292      0.754   
4    0.298      0.733  0.424      0.728        100  0.795   0.312      0.786   
5    0.444      0.923  0.600      0.867        125  0.892   0.617      0.963   
6    0.417      0.860  0.562      0.862        128  0.898   0.631      0.979   
7    0.390      0.885  0.541      0.833        160  0.872   0.532      0.975   
8    0.315      0.894  0.466      0.820        200  0.860   0.492      0.967   
9    0.346      0.911  0.501      0.862        250  0.891   0.603      0.978   
10   0.258      0.905  0.401      0.808        320  0.881   0.556      0.994   
11   0.214      0.829  0.340      0.768        400  0.859   0.475      0.993   
12   0.203      0.779  0.323      0.745        500  0.852   0.451      0.985   
13   0.234      0.896  0.371      0.826        640  0.847   0.444      0.956   
14   0.231      0.907  0.368      0.818        800  0.834   0.390      0.966   
15   0.234      0.896  0.371      0.804       1000  0.837   0.403      0.960   
16   0.214      0.900  0.345      0.759       1600  0.849   0.447      0.964   
17   0.200      0.894  0.327      0.775       2000  0.841   0.407      0.992   
18   0.183      0.900  0.304      0.755       3200  0.841   0.410      0.976   
19   0.176      0.881  0.294      0.752       4000  0.843   0.414      0.992   
20   0.166      0.980  0.284      0.747       8000  0.831   0.366      0.991   
21   0.176      0.929  0.296      0.750      10000  0.838   0.393      0.991   
22   0.173      0.981  0.294      0.736      13000  0.837   0.400      0.967   

       f1  auc_score  
0   0.185      0.553  
1   0.445      0.752  
2   0.455      0.744  
3   0.421      0.750  
4   0.447      0.749  
5   0.752      0.940  
6   0.767      0.936  
7 

In [16]:
res_result = pd.concat([res_type_6_csv, res_type0_csv, res_type6_csv], axis = 1)
res_result

res_type    acc  recall  precision     f1  auc_score  \
0   abs_max(our_custom)  0.741   0.200      0.536  0.291      0.613   
1           kaiser_best  0.782   0.264      0.757  0.392      0.726   
2           kaiser_fast  0.783   0.275      0.750  0.402      0.726   
3                   fft  0.777   0.244      0.742  0.367      0.697   
4                 scipy  0.777   0.244      0.742  0.367      0.697   
5             polyphase  0.768   0.231      0.694  0.346      0.707   
6                linear  0.717   0.058      0.321  0.098      0.530   
7       zero_order_hold  0.717   0.058      0.321  0.098      0.530   
8             sinc_best  0.734   0.000      0.000  0.000      0.500   
9           sinc_medium  0.734   0.000      0.000  0.000      0.500   
10         sinc_fastest  0.734   0.000      0.000  0.000      0.500   
11             soxr_vhq  0.781   0.241      0.789  0.369      0.708   
12              soxr_hq  0.779   0.241      0.772  0.367      0.720   
13              soxr_mq  0.784   0.251      0.796  0.381      0.722   
14              soxr_lq  0.771   0.268      0.675  0.383      0.718   
15              soxr_hq  0.779   0.241      0.772  0.367      0.720   
16              soxr_qq  0.720   0.088      0.382  0.143      0.559   

              target_sr    acc  recall  precision     f1  auc_score  \
0   abs_max(our_custom)  0.811   0.376      0.810  0.514      0.690   
1           kaiser_best  0.842   0.444      0.923  0.600      0.867   
2           kaiser_fast  0.853   0.485      0.929  0.637      0.873   
3                   fft  0.826   0.407      0.870  0.554      0.836   
4                 scipy  0.826   0.407      0.870  0.554      0.836   
5             polyphase  0.834   0.431      0.888  0.580      0.853   
6                linear  0.778   0.231      0.782  0.356      0.689   
7       zero_order_hold  0.778   0.231      0.782  0.356      0.689   
8             sinc_best  0.838   0.417      0.939  0.577      0.860   
9           sinc_medium  0.837   0.431      0.907  0.584      0.867   
10         sinc_fastest  0.848   0.461      0.932  0.617      0.868   
11             soxr_vhq  0.841   0.441      0.922  0.596      0.862   
12              soxr_hq  0.841   0.444      0.916  0.598      0.859   
13              soxr_mq  0.842   0.437      0.935  0.596      0.860   
14              soxr_lq  0.852   0.478      0.934  0.632      0.875   
15              soxr_hq  0.841   0.444      0.916  0.598      0.859   
16              soxr_qq  0.769   0.214      0.724  0.330      0.665   

              target_sr    acc  recall  precision     f1  auc_score  
0   abs_max(our_custom)  0.829   0.410      0.883  0.560      0.761  
1           kaiser_best  0.892   0.617      0.963  0.752      0.940  
2           kaiser_fast  0.888   0.603      0.962  0.742      0.941  
3                   fft  0.880   0.573      0.960  0.718      0.933  
4                 scipy  0.880   0.573      0.960  0.718      0.933  
5             polyphase  0.888   0.600      0.967  0.741      0.932  
6                linear  0.825   0.380      0.911  0.536      0.765  
7       zero_order_hold  0.825   0.380      0.911  0.536      0.765  
8             sinc_best  0.889   0.593      0.983  0.740      0.935  
9           sinc_medium  0.893   0.610      0.978  0.752      0.941  
10         sinc_fastest  0.886   0.607      0.942  0.738      0.931  
11             soxr_vhq  0.886   0.586      0.972  0.732      0.935  
12              soxr_hq  0.885   0.593      0.956  0.732      0.939  
13              soxr_mq  0.885   0.600      0.947  0.734      0.937  
14              soxr_lq  0.898   0.641      0.964  0.770      0.937  
15              soxr_hq  0.885   0.593      0.956  0.732      0.939  
16              soxr_qq  0.820   0.386      0.857  0.533      0.783

# sr결과

- noise_6 : f1 = 64, auc = 64
- noise0 : 1,2등 둘다 (125, 128) 둘다해보기 
- noise6 : f1 = 128, auc = 250